app/models/post.rb

```diff
@@ -1,4 +1,4 @@
 class Post < ActiveRecord::Base
-  belongs_to :charity
+  belongs_to :charity, touch: true
```

Comment:
adding touch: true tells helps the solr server know that when a charity creates a post, it will need to do something. In this case, on the Charity model I added `after_touch :index` so that the post model (title and content) will be reindexed and searchable after the post is 'touched' (saved or edited).